lib/trino-orc/src/main/java/io/trino/orc/metadata/OrcMetadataReader.java

```
@@ -150,7 +153,8 @@ public Footer readFooter(HiveWriterVersion hiveWriterVersion, InputStream inputS
                 toType(footer.getTypesList()),
                 toColumnStatistics(hiveWriterVersion, footer.getStatisticsList(), false),
                 toUserMetadata(footer.getMetadataList()),
-                Optional.of(footer.getWriter()));
+                Optional.of(footer.getWriter()),
+                Optional.of(toTrinoOrcCalendarKind(footer.getCalendar())));
```

Should we check if it is `PROLEPTIC_GREGORIAN` when reading?
we can throw exception for `JULIAN_GREGORIAN` since we don't support, and what if it is `UNKNOWN_CALENDAR`?

---

Yes, until such a time when we support Julian, we should throw an error -- otherwise, the reader will produce incorrect values.

---

I'm strongly against 1. Changing the behaviour of the reader to now starting throwing exceptions for all dates if we don't recognize calendar type could be highly disruptive to existing workloads.
2 would be ideal, but I don't recommend spending time on that. Over time workloads are moving away from Hive and into iceberg/delta, so we shouldn't spend too much time agonizing over adjusting to Hive intricacies.
You can file a separate issue about the reader part, I don't see why that should block the writer part in this PR.

---

@mosabua 
Which place in the doc could be the best place to put info about the limitation of ORC reader for legacy dates and timestamp when Calendar in the Orc file is set to Hybrid. or maybe the issue is just enough  (https://github.com/trinodb/trino/issues/26865)

---

I think probably in https://trino.io/docs/current/object-storage/file-formats.html#orc-format-configuration-properties somehow

---

lgtm, please resolve the build failures